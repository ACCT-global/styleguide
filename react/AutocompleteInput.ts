export { default } from './components/Autocomplete'
