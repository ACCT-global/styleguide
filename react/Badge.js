import Tag from './components/Tag/index'
import deprecated from './modules/deprecated'

export default deprecated(
  {
    useNewComponent: {
      Badge: 'Tag',
    },
  },
  Tag
)
