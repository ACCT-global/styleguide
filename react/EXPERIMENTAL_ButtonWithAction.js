import EXPERIMENTAL_ButtonWithAction from './components/EXPERIMENTAL_ButtonWithAction/index'

export default EXPERIMENTAL_ButtonWithAction
