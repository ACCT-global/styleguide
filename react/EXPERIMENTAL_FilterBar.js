import EXPERIMENTAL_FilterBar from './components/EXPERIMENTAL_FilterBar/index'

export default EXPERIMENTAL_FilterBar
