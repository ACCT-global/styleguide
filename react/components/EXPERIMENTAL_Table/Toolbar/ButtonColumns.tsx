import React, { useMemo, FC } from 'react'

import Toggle from '../../../Toggle'
import IconColumns from '../../icon/Columns/index'
import Menu from './Menu/index'

import { ICON_SIZE, COLUMNS_BOX, NAMESPACES } from '../constants'
import useTableContext from '../hooks/useTableContext'

export type ButtonColumnsProps = {
  label: string
  showAllLabel: string
  hideAllLabel: string
  alignMenu: Alignment
  disabled: boolean
}

const ButtonColumns: FC<ButtonColumnsProps> = ({
  label,
  showAllLabel,
  hideAllLabel,
  alignMenu,
  disabled,
}) => {
  const {
    hiddenColumns,
    columns,
    hideAllColumns,
    showAllColumns,
    toggleColumn,
  } = useTableContext()

  const calculateFieldsBoxHeight = () => {
    const estimate = Object.keys(columns).length * COLUMNS_BOX.ITEM_HEIGHT
    return estimate > COLUMNS_BOX.MAX_HEIGHT ? COLUMNS_BOX.MAX_HEIGHT : estimate
  }

  const height = useMemo(() => calculateFieldsBoxHeight(), [
    Object.keys(columns).length,
  ])

  return (
    <Menu
      button={{
        id: NAMESPACES.TOOLBAR.BUTTON_COLUMNS,
        title: label,
        icon: <IconColumns size={ICON_SIZE.MEDIUM} />,
        disabled: disabled,
      }}
      box={{
        height,
        alignMenu,
        width: COLUMNS_BOX.WIDTH,
        groupActions: [
          { id: 1, label: showAllLabel, onClick: showAllColumns },
          { id: 2, label: hideAllLabel, onClick: hideAllColumns },
        ],
      }}>
      {columns.map((column, index) => {
        const { id, title } = column
        const togglerFn = () => toggleColumn(id)
        const isVisible = !hiddenColumns.includes(id)
        return (
          <Menu.Item key={index} handleCallback={togglerFn}>
            {title}
            <Toggle checked={isVisible} onChange={togglerFn} />
          </Menu.Item>
        )
      })}
    </Menu>
  )
}

export default ButtonColumns
