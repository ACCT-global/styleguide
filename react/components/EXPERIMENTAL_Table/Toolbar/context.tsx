import React, { FC, useContext, createContext } from 'react'

import { E2ETestable } from '../types'

const ToolbarContext = createContext<E2ETestable>(null)

export const ToolbarProvider: FC<E2ETestable> = ({ children, testId }) => {
  return (
    <ToolbarContext.Provider value={{ testId }}>
      {children}
    </ToolbarContext.Provider>
  )
}

export function useToolbarContext() {
  const context = useContext(ToolbarContext)
  if (!context) {
    throw new Error('Do not use Toolbar composites outside of context')
  }
  return context
}
