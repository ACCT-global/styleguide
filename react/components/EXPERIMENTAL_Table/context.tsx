import React, { FC, useContext, createContext } from 'react'

import { E2ETestable } from './types'
import useTableMeasures from './hooks/useTableMeasures'

type Measures = ReturnType<typeof useTableMeasures>

const TestingContext = createContext<E2ETestable>(null)
const MeasuresContext = createContext<Measures>(null)

type TableType = E2ETestable & {
  measures: Measures
}

export const TableProvider: FC<TableType> = ({
  children,
  testId,
  measures,
}) => {
  return (
    <TestingContext.Provider value={{ testId }}>
      <MeasuresContext.Provider value={measures}>
        {children}
      </MeasuresContext.Provider>
    </TestingContext.Provider>
  )
}

export function useTestingContext() {
  const context = useContext(TestingContext)
  if (!context) {
    throw new Error(
      'Do not use E2ETestable components outside of the TestingContext'
    )
  }
  return context
}

export function useMeasuresContext() {
  const context = useContext(MeasuresContext)
  if (!context) {
    throw new Error(
      'Do not use measurable components outside of the MeasuresContext'
    )
  }
  return context
}
