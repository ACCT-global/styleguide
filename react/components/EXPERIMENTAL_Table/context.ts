import { createContext } from 'react'

const Context = createContext<TableState>(null)
const { Provider } = Context

export default Context
export { Provider as TableProvider }
