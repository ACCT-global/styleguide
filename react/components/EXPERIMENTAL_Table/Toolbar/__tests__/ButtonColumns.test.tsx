import React from 'react'
import { renderHook } from '@testing-library/react-hooks'
import { render, fireEvent, screen } from '@testing-library/react'

import ButtonColumns from '../ButtonColumns'
import { ButtonGroupProvider } from '../context'
import useTableVisibility from '../../hooks/useTableVisibility'

describe('Table V2 @ Toolbar/ButtonColumns spec', () => {
  const basicProps = {
    label: 'Toggle visible fields',
    showAllLabel: 'Show All',
    hideAllLabel: 'Hide All',
  }

  const { result } = renderHook(() =>
    useTableVisibility({
      columns: [{ id: 'name', title: 'Name' }],
      hiddenColumns: [],
    })
  )

  it('matches snapshot', () => {
    const button = render(
      <ButtonGroupProvider testId="testId">
        <ButtonColumns visibility={result.current} {...basicProps} />
      </ButtonGroupProvider>
    )
    expect(button.asFragment()).toMatchSnapshot()
  })
  it('matches snapshot after click', () => {
    const button = render(
      <ButtonGroupProvider testId="testId">
        <ButtonColumns visibility={result.current} {...basicProps} />
      </ButtonGroupProvider>
    )
    fireEvent.click(screen.getByTestId('testId__columns'))
    expect(button.asFragment()).toMatchSnapshot()
  })
})
