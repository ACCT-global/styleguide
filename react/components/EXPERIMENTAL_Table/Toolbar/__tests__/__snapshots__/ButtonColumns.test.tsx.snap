// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table V2 @ Toolbar/ButtonColumns spec matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative mh2"
    data-testid="testId__columns__container"
    id="vtex-table-v2__toolbar__button-columns__container"
    title="Toggle visible fields"
  >
    <button
      class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small icon-button dib bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary pointer "
      data-testid="testId__columns"
      style="font-size: 0px;"
      tabindex="0"
      type="button"
    >
      <div
        class="vtex-button__label flex items-center justify-center h-100 ph3 "
        style="padding-top: .25em; padding-bottom: .32em;"
      >
        <span
          class="c-on-base"
          style="margin-top: 1.5px;"
        >
          <svg
            class="vtex__icon-columns  "
            fill="none"
            height="14"
            viewBox="0 0 16 16"
            width="12.444444444444445"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <path
              d="M1 4H15V15H1V4Z"
              stroke="currentColor"
              stroke-width="2"
            />
            <path
              d="M0 1C0 0.447715 0.447715 0 1 0H15C15.5523 0 16 0.447715 16 1V5H0V1Z"
              fill="currentColor"
            />
            <line
              stroke="currentColor"
              stroke-width="2"
              x1="8.11133"
              x2="8.11133"
              y1="4.70586"
              y2="14.1176"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Table V2 @ Toolbar/ButtonColumns spec matches snapshot after click 1`] = `
<DocumentFragment>
  <div
    class="relative mh2"
    data-testid="testId__columns__container"
    id="vtex-table-v2__toolbar__button-columns__container"
    title="Toggle visible fields"
  >
    <button
      class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small icon-button dib bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary pointer "
      data-testid="testId__columns"
      style="font-size: 0px;"
      tabindex="0"
      type="button"
    >
      <div
        class="vtex-button__label flex items-center justify-center h-100 ph3 "
        style="padding-top: .25em; padding-bottom: .32em;"
      >
        <span
          class="c-on-base"
          style="margin-top: 1.5px;"
        >
          <svg
            class="vtex__icon-columns  "
            fill="none"
            height="14"
            viewBox="0 0 16 16"
            width="12.444444444444445"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <path
              d="M1 4H15V15H1V4Z"
              stroke="currentColor"
              stroke-width="2"
            />
            <path
              d="M0 1C0 0.447715 0.447715 0 1 0H15C15.5523 0 16 0.447715 16 1V5H0V1Z"
              fill="currentColor"
            />
            <line
              stroke="currentColor"
              stroke-width="2"
              x1="8.11133"
              x2="8.11133"
              y1="4.70586"
              y2="14.1176"
            />
          </svg>
        </span>
      </div>
    </button>
    <div
      class="absolute z-999 shadow-4 b--muted-4 br2 ba left-0 mt2 mh2"
      data-testid="testId__columns__box"
      style="width: 292px;"
    >
      <div
        class="w-100 b2 br2 bg-base"
      >
        <div
          class="flex inline-flex bb b--muted-4 w-100 justify-center pv4"
          data-testid="testId__columns__box__group-actions"
        >
          <div
            class="mh2"
            data-testid="testId__columns__group-actions--1"
          >
            <button
              class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-action-secondary b--action-secondary c-on-action-secondary hover-bg-action-secondary hover-b--action-secondary hover-c-on-action-secondary pointer "
              tabindex="0"
              type="button"
            >
              <div
                class="vtex-button__label flex items-center justify-center h-100 ph5 "
                style="padding-top: .25em; padding-bottom: .32em;"
              >
                Show All
              </div>
            </button>
          </div>
          <div
            class="mh2"
            data-testid="testId__columns__group-actions--2"
          >
            <button
              class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-action-secondary b--action-secondary c-on-action-secondary hover-bg-action-secondary hover-b--action-secondary hover-c-on-action-secondary pointer "
              tabindex="0"
              type="button"
            >
              <div
                class="vtex-button__label flex items-center justify-center h-100 ph5 "
                style="padding-top: .25em; padding-bottom: .32em;"
              >
                Hide All
              </div>
            </button>
          </div>
        </div>
        <div
          class="overflow-auto"
          data-testid="testId__columns__box__items"
          style="height: 36px;"
        >
          <div
            class="flex justify-between ph6 pv3 pointer hover-bg-muted-5 bl bw1 b--transparent"
          >
            <span
              class="w-100 flex justify-between"
            >
              Name
              <label>
                <div
                  class="flex flex-row items-center relative pointer"
                >
                  <div
                    class="vtex-toggle flex items-center relative br4 bg-action-primary "
                    style="transition: background 100ms ease-out; height: 1.25rem; width: 2.25rem; min-width: 2.25rem;"
                  >
                    <div
                      class="absolute br-100 bg-base "
                      style="box-shadow: 0 0 10px rgba(0,0,0,0.2); transform: scale(0.8); transition: all .1s ease-out; height: 1.25rem; width: 1.25rem; min-width: 1.25rem; left: 1rem;"
                    />
                  </div>
                  <input
                    checked=""
                    class="h-100 w-100 absolute o-0 pointer"
                    tabindex="0"
                    type="checkbox"
                  />
                </div>
              </label>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
